src/mapgen/mg_schematic.cpp

```diff
@@ -82,7 +82,7 @@ void Schematic::resolveNodeNames()
 {
 	getIdsFromNrBacklog(&c_nodes, true, CONTENT_AIR);
 
-	size_t bufsize = size.X * size.Y * size.Z;
+	size_t bufsize = (int)size.X * size.Y * size.Z;
```

Comment:
What's the warning here? Integer overflow?

It's probably not very useful to fix this here because a schematic that overflows the size calculation will cause bugs in other places anyway.

Reply:
https://lgtm.com/projects/g/minetest/minetest/snapshot/edcf968aa097bcd285abc86134a9561f9bd43bac/files/src/mapgen/mg_schematic.cpp

Reply:
Have you considered using a static_cast for this?